### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by phalado**

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,65 +1,48 @@
...
-      @player2 = gets.chomp
+      @engine.player2 = gets.chomp
       puts `clear`
     rescue StandardError => e
-      p "No name given"
+      p "No name given #{e}"
+      sleep 1
       retry
-    else
     end
```
Comment (maelfosso) [inline bin/main]: According to that, it's possible for you to initialize the TicTacLogic instance with players as parameters.
Why not do that ?

File: bin/main [hunk 2]
```diff
@@ -1,65 +1,48 @@
...
-    moves = 0
     paint
-    while @play
-      p "#{ @playert ? @player1 : @player2 }, make a move"
-      begin
-        move((gets.chomp.to_i) - 1)
-      rescue StandardError => e
-        p "Error!!"
-        retry
+    p "Please #{@engine.playert ? @engine.player1 : @engine.player2}, make a move"
```
Comment (maelfosso) [inline bin/main]: Define a method for that
```ruby
def current_player
   @engine.playert ? @engine.player1 : @engine.player2
end
```

Something I like that

File: lib/logic.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class TicTacLogic
+  attr_accessor :player1, :player2
```
Comment (maelfosso) [inline lib/logic.rb]: `attr_accessor` is not allowed, you should use either `attr_reader` or `attr_writer`

File: lib/logic.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  attr_reader :board, :play, :playert
+
+  def initialize
+    @play = true
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @playert = true
+  end
+
+  def move(num)
+    if @board[num - 1].is_a?(Integer) && num >= 0 && num <= 9
```
Comment (maelfosso) [inline lib/logic.rb]: Usually it recommended to check for index validity before the value into the array at that index position
Not that is not working but it's not a good habit
```ruby
if num >= 0 && num <= 9 && @board[num - 1].is_a?(Integer) 
```
is preferable.

Take the habit to write it like this

File: lib/logic.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      @board[num - 1] = @playert ? 'X' : 'O'
```
Comment (maelfosso) [inline lib/logic.rb]: Please, use symbol `:X` or `:O` than character

File: lib/logic.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      @play = false
+      "Game over - Winner: #{@playert ? @player1 : @player2}"
+    elsif @board.none?(Integer)
+      @play = false
+      "Game over - it's a tie"
+    else
+      @playert = !@playert
+      "Please #{@playert ? @player1 : @player2}, make a move"
+    end
+  end
```
Comment (maelfosso) [inline lib/logic.rb]: Returning string is not a good habit
Please, return a value and use that value to decided what to print

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @phalado @Chincoya 

Nice game :clap:  !!! Simple and amazing :smile: !!
I have nothing to say about but there are some changes that I suggest concerning your code.
Please, read the comments to know more about them.

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements so I am approving your project. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Pull request merged**